docs/website/docs/developers/general/github-actions.md

```diff
@@ -0,0 +1,290 @@
+[`samples.yml`](https://github.com/iree-org/iree/blob/main/.github/workflows/samples.yml) | [![Samples](https://github.com/iree-org/iree/actions/workflows/samples.yml/badge.svg?query=branch%3Amain+event%3Aschedule)](https://github.com/iree-org/iree/actions/workflows/samples.yml?query=branch%3Amain+event%3Aschedule) | `schedule`
+
+## :octicons-pencil-16: Writing and editing workflows
+
+### :simple-docker: Docker and dependencies
+
+Workflow files typically require some external dependencies in the form of
+software packages, environment settings, and sometimes even system/hardware
+drivers. One way to manage these dependencies is to bundle them into a container
+using a tool like [Docker](https://www.docker.com/).
+
+!!! tip
+
+    We recommend only using Docker containers within workflow files in specific
+    cicumstances and with moderation.
+
+    IREE contains a cross-compiler and minimal runtime, both of which are
+    designed to run on a wide range of systems. Using carefully constructed
+    containers for basic development risks the project _only_ working within
+    such containers.
+
+These sorts of dependencies _may_ be a good fit for using Docker containers:
+
+* Infrequently changing large dependencies like compiler toolchains.
+* Dependencies with complicated installs (e.g. building from source, moving
+  files to specific paths).
+* System dependencies like GPU drivers.
+* Environment settings like disk partitions.
+
+Here are alternative ways to fetch and configure workflow/job dependencies:
+
+* Install from a package manager like `pip` or `apt`.
+* Use an action like
+  [actions/setup-python](https://github.com/actions/setup-python) to install
+  packages and add them to `PATH`.
+* Use GitHub-hosted runners and their
+  [installed software](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#installed-software).
+
+### :octicons-workflow-16: Workflow triggers
+
+Of the
+[events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows),
+we most commonly use:
+
+* `pull_request`
+    * Jobs most representative of core developer workflows should aim to run
+      here. Jobs can be marked [required](#required-and-optional-checks) or
+      [opt-in on presubmit](#opt-in-for-presubmit-jobs) on a case-by-case basis.
+* `push`
+    * Jobs running here should be a superset of jobs running on `pull_request`.
+* `schedule`
+    * Jobs designed to run nightly (e.g. nightly releases), jobs for non-core
+      configurations (like certain sanitizers/fuzzers), and jobs using
+      self-hosted runners in low supply can run on a schedule instead of on
+      every commit.
```

Comment:
I could add a few examples here explaining the current state.

* Why ci_linux_x64_clang_asan.yml is running on `pull_request` but ci_linux_x64_clang_tsan.yml is running on `schedule` (asan is extremely useful, tsan is only situationally useful, both require big CPU runners)
* Why ci_windows_x64_msvc.yml is running on `schedule` (standard GitHub-hosted runners are slow, large GitHub-hosted runners are expensive, our CPU builder pool only has Linux runners in it)
* Why ci_linux_arm64_clang.yml is running on `schedule` (Needs either large GitHub-hosted runners or self-hosted, and we had availability issues with the self-hosted runners)